spec/unit/lib/temporal/workflow/state_manager_spec.rb

```diff
@@ -36,4 +38,30 @@ class MyWorkflow < Temporal::Workflow; end
       end
     end
   end
-end
\ No newline at end of file
+
+  describe '#search_attributes' do
+    let(:start_workflow_execution_event) { Fabricate(:api_workflow_execution_started_event) }
```

Comment:
not sure how common it will be, but you should probably test a third, overwritten search attribute as well to demonstrate the upsert semantics.